ERCS/erc-7689.md

```diff
@@ -0,0 +1,268 @@
+
+#### blobVM Context
+
+The blobVM context is injected by the sequencer during the lazy evaluation of a transaction. Below are the mandatory methods in the Context:
+
+| method  | description |
+| :-------------: |:-------------:|
+| `blobvm.msg.sender` | return the transaction sender (EOA)     |  
+| `blobvm.tx.id`      | return the call's transaction id     |  
+
+
+#### Gas formula
+A blobVM transaction consists of two factors affecting the gas calculation: gas paid for the EVM layer 1 (L1) and gas paid to the blobVM sequencer (Sequencer):
+
+- **Gas Paid for L1**: This is the gas paid by a transaction that implements [EIP-1559](./eip-1559.md) and [EIP-4844](./eip-4844.md) standards.
+
+- **Gas Paid to the Sequencer**: This occurs within the same transaction. It involves transferring a sufficient amount of gas fee to the sequencer's address under the `to` (destination) field.
+
+The gas cost of a blobVM transaction (types `1` and `2`) is calculated as follows:
+
+```plaintext
+tx_gas = l1_gas_fees + (262604 * winston_byte_price * 1e-12 * ar_usd_price / eth_usd_price) * bvm_multiplier
+```
+
+#### Equation Terms Breakdown:
+
+- `l1_gas_fees`: The gas paid to post the transaction to the EVM network.
+- `262604`: The total byte size of an `EIP-4844` transaction when archiving on Arweave. This includes data, KZG commitments, and proof.
+- `winston_byte_price`: The cost price per byte on Arweave. This is dynamic and can be checked at `https://arweave.net/price/262604`.
+- `1e-12 * ar_usd_price`: The conversion of `winston_byte_price` from winstons to AR and then to USD.
+- `sequencer_multiplier` (>= 1): The total Arweave cost, converted to ETH, is then multiplied by the sequencer premium multiplier.
+
+### Gateway interface
+The designed sequencer MUST consistently expose the following methods:
+
+#### Reading contract state
+
+```bash
+curl -X GET base_endpoint/state/target_contract_addr
+```
+
+#### Deploying a contract
+
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"txid": "contract_eip4844_txid"}' base_endpoint/deploy 
+```
+
+#### Sending a transaction
+
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"txid": "eip4844_txid"}' base_endpoint/transactions 
+```
+***N.B: Smart Blobs sequencers are centralized and do not offer censorship resistance guarantees. The end user must rely on the good faith of the sequencer.***
+
+## Rationale
+This proposal hopes to accomplish the following:
+* Standardizing the usage of EIP-4844 blobs within the scope of creating execution machines on top of the blobspace
+* Utilize SmartWeave's lazy evaluation paradigm within the EVM space.
+* Extend the capabilities of lazy evaluation by using window-timegated DA
+* Utilize blobspace instead of calldata to leverage off-chain execution more cheaply
+* Write Smart Blobs in any interpreted language, or in a compiled language provided it uses WASM (JavaScript, Python, Rust, Lua, etc.)
```

Comment:
The Rationale section is used to explain choices made in the proposal, while this content seems to be justifying the EIP as a whole (and so belongs in the Motivation section.) My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We painted the shed red because...